### ISSUE TITLE
systemd input not working (Ubuntu)

### ISSUE THREAD

**Issue by uninvited** [fluent/fluent-bit#2525]
## Bug Report
I'm trying to use systemd input on Ubuntu 18.04, but get an error

> Input plugin 'systemd' cannot be loaded
> Error: Configuration file contains errors. Aborting

The configuration is similar to the one from the examples:
```
[SERVICE]
    Flush        1
    Log_Level    info
    Parsers_File parsers.conf

[INPUT]
    Name            systemd
    Tag             host.*
    Systemd_Filter  _SYSTEMD_UNIT=cron.service

[OUTPUT]
    Name   stdout
    Match  *

```

I've tried to re-build with `-DFLB_IN_SYSTEMD=On` (though it seems to be enabled by default anyway), without success.
These packages are installed:

> ii  libsystemd-dev:amd64         237-3ubuntu10.42                                amd64        systemd utility library - development files
> ii  libsystemd0:amd64              237-3ubuntu10.42                                amd64        systemd utility library
> ii  libsystemd0:i386                  237-3ubuntu10.42                                i386         systemd utility library

I've also tried to install systemd plugin separately:

> gem install fluent-plugin-systemd -v 1.0.1

but it did not help either.

* Version used: 1.6.0, built from source
* Operating System and version: Ubuntu 18.04

**Comment by PettitWesley** [comment]
> gem install fluent-plugin-systemd -v 1.0.1

Anything you can `gem install` is Ruby and thus is Fluentd. 

I had the same problem on Amazon Linux a long time ago. You must install the right systemd packages, the ones for development. On some distros they will have "devel" in the name. Not sure what the right one is one Ubuntu. 

The dockerfile here is based on debian, package names might be the same on ubuntu: https://github.com/fluent/fluent-bit-docker-image/blob/1.5/Dockerfile.x86_64

**Comment by uninvited** [comment]
Wesley, thanks for the reply!
> You must install the right systemd packages, the ones for development. On some distros they will have "devel" in the name. Not sure what the right one is one Ubuntu.

I have `libsystemd-dev` and `libsystemd0` packages installed. Could not find any other systemd-related packages for Ubuntu.
Also, just checked that I do have these files on my system (they are mentioned in your Dockerfile):
> /lib/x86_64-linux-gnu/libsystemd.so
> /lib/x86_64-linux-gnu/libsystemd.so.0
> /lib/x86_64-linux-gnu/libsystemd.so.0.21.0

So maybe systemd input plugin is just not supported on Ubuntu? Has it been tested with this OS?

Actually my final goal is to use fluent-bit on embedded linux distribution built with yocto. I tested it on Ubuntu just to familiarize myself with fluent-bit. (By the way, systemd input plugin did not work on my embedded device either).

**Comment by PettitWesley** [comment]
Systemd input plugin definitely works on Ubuntu. I mainly use Amazon Linux as my testing/dev enviroment, but I also have an Ubuntu VM which I occasionally use. I have built Fluent Bit from source on it. I just checked, the systemd plugin seems to have been built fine. I can reference it in a config. I don't get the "Input plugin 'systemd' cannot be loaded" message on startup.

And the main Fluent Bit docker image is built on debian, from which Fluent Bit is derived...

> By the way, systemd input plugin did not work on my embedded device either

I'm not an expert on these things... but to me it feels like systemd wouldn't be used on an embedded device anyway?
